gitlab/client.py

```diff
@@ -511,27 +511,21 @@ def _set_auth_info(self) -> None:
             not self.http_username and self.http_password
         ):
             raise ValueError("Both http_username and http_password should be defined")
-        if self.oauth_token and self.http_username:
+        if len(tokens) and self.http_username:
             raise ValueError(
-                "Only one of oauth authentication or http "
+                "Only one of token authentications or http "
                 "authentication should be defined"
             )
 
-        self._http_auth = None
+        self._http_auth: Optional[requests.auth.AuthBase] = None
         if self.private_token:
-            self.headers.pop("Authorization", None)
-            self.headers["PRIVATE-TOKEN"] = self.private_token
-            self.headers.pop("JOB-TOKEN", None)
+            self._http_auth = _backends.HTTPPrivateTokenAuth(self.private_token)
 
         if self.oauth_token:
-            self.headers["Authorization"] = f"Bearer {self.oauth_token}"
-            self.headers.pop("PRIVATE-TOKEN", None)
-            self.headers.pop("JOB-TOKEN", None)
+            self._http_auth = _backends.HTTPOAuthTokenAuth(self.oauth_token)
 
         if self.job_token:
-            self.headers.pop("Authorization", None)
-            self.headers.pop("PRIVATE-TOKEN", None)
-            self.headers["JOB-TOKEN"] = self.job_token
+            self._http_auth = _backends.HTTPJobTokenAuth(self.job_token)
 
         if self.http_username and self.http_password:
             self._http_auth = requests.auth.HTTPBasicAuth(
```

Comment:
Similar to the above - now that we encapsulate all types of auth here, I would maybe just call it `self._auth` for brevity.

In fact, the current `_http_auth` based on `HTTPBasicAuth` is dead code because GitLab hasn't supported basic auth with user/password for years, so we can remove it in a follow-up.

Reply:
Makes sense, implemented